lib/private/Files/ObjectStore/S3ConnectionTrait.php

```diff
@@ -81,6 +84,7 @@ protected function parseParams($params) {
 		$this->bucket = $params['bucket'];
 		$this->proxy = $params['proxy'] ?? false;
 		$this->timeout = $params['timeout'] ?? 15;
+		$this->storageClass = (isset($params['storageClass']) && !empty($params['storageClass'])) ? $params['storageClass'] : 'STANDARD';
```

Comment:
Why not use the same logic as for other params ?

```suggestion
		$this->storageClass = $params['storageClass'] ?? 'STANDARD';
```

Reply:
That's what I've done at start ;-)

But in the external storage application (files_external), if the storage class is not specified, you get an empty class (an empty string), which is not a valid a storage class for the providers and generates an error. Not sure you can specified a default value for `OCA\Files_External\Lib\DefinitionParameter`?

It seemed to me more generic to ensure here that, if the storage class is not defined or is empty, we fallback to *STANDARD*.

Reply:
```suggestion
		$this->storageClass = (!empty($params['storageClass'])) ? $params['storageClass'] : 'STANDARD';
```

Then no need for `isset`, `empty` implies `isset`.

Reply:
ok, the PR have been updated